### PR TITLE
feat: add "somewhere else" as a location option

### LINKED ISSUE #14219: [CAL-3855] Feature Request: "Somewhere Else"
I would like to move from Calendly.com to Cal.com so I can support a great open source product but there is an option that exists in Calendly that does not exist in Cal.com. 

The option to add "Ask Invitee" that has the Verbiage  "Somewhere else" as the field name on the booking page and "Any location" as the location suggestion text:

image

I say this differs from "In-Person (Attendee Address)" because the alternate location I want them to put is possibly _their_ Teams, Google Meet, or Zoom link in their tenant. Or possibly an 8x8 or RingCentral Meeting link or another "location" of their choice in the virtual world. But also "My Office Conference Room" could also fit in this Somewhere Else/Any Location option field.

I hope I was clear in my feature request and my reasoning behind it. I appreciate all of you and all the work you all do.

<sub>[CAL-3855](https://linear.app/calcom/issue/CAL-3855/feature-request-somewhere-else)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

- Fixes #14219 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx
Comment (sean-brydon) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:88]: I've just noticed something - that is in this PR and also main 

locationResponse?.value === "attendeeInPerson" || "phone" || "somewhereElse"

If `locationResponse?.value === "attendeeInPerson" ` is false we goto "phone"

Which in theory returns true even when we might not want to 

`Boolean("phone") === true` I think we need to do something simular to the files above  `apps/web/components/dialog/EditLocationDialog.tsx` to check if it is in the array

**Review by sean-brydon — DISMISSED** [review]
I think we should confirm the change in `packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx` is doing what we expect

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx
Comment (emrysal) [reply packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:88]: Yeah, and also let's add a test to confirm this works (obviously this is entirely uncovered or otherwise it'd be exposed earlier)

**Comment by anikdhabal** [comment]
@sean-brydon @emrysal thanks for the review. I will update that shortly.

**Review by Amit91848 — DISMISSED** [review]
LGTM and works great @anikdhabal , ~~but we should add a test as suggested here https://github.com/calcom/cal.com/pull/15108#discussion_r1608367748 before merging this~~

Rescheduling issues are already being dealt with in #15810 , should be good to go once that is merged

**Comment by anikdhabal** [comment]
This [comment](https://github.com/calcom/cal.com/pull/15108#discussion_r1608367748) issue is addresed by hariom in this pr https://github.com/calcom/cal.com/pull/16162

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
'Somewhere Else' is not a good label for the booking page if it's the only option:
Screenshot 2024-08-28 at 11 20 14 AM

Somewhere else only makes sense if the attendee can choose between different location options. If it's the only option we should probably just label it as Location 


NIT: there is too much space between label and input field (issue already existed before, so can be fixed in a separate PR)

**Comment by CarinaWolli** [comment]
Booking page looks good now 👍 Could we also rename it in the event type settings? 'Somewhere else' is also not really clear there, maybe we call it 'Custom attendee location', but I am open to other suggestions

**Comment by anikdhabal** [comment]
> Booking page looks good now 👍 Could we also rename it in the event type settings? 'Somewhere else' is also not really clear there, maybe we call it 'Custom attendee location', but I am open to other suggestions

Hey @CarinaWolli thanks for the suggestion, but what about `Any Location`? Both in booking and event type page settings

**Comment by CarinaWolli** [comment]
> Hey @CarinaWolli thanks for the suggestion, but what about `Any Location`? Both in booking and event type page settings

I like how the booking page is now with 'Somewhere else' and 'Location', so I'd prefer to leave that and find a better name for the event-type settings

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
It shows custom_attendee_location here instead of the translated value, everything else looks good!
Screenshot 2024-08-30 at 1 09 33 PM

**Comment by anikdhabal** [comment]
Now everything is fine
Screenshot 2024-08-30 230455

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**
